compiler/passes/scopeResolve.cpp

```diff
@@ -1703,136 +1716,288 @@ static void    buildBreadthFirstModuleList(Vec<ModuleSymbol*>* modules,
                                            Vec<ModuleSymbol*>* current,
                                            Vec<ModuleSymbol*>* alreadySeen);
 
-static Symbol* lookup(BaseAST* scope, const char* name)
-{
-  Vec<Symbol*>  symbols;
+// Given a name and a scope, determine the symbol referred by that name in the
+// context of that scope.
+static Symbol* lookup(BaseAST* scope, const char* name) {
+  Symbol * symbolResult;
```

Comment:
It might be preferred to initialize symbolResult to NULL to ensure that this function never
returns garbage.  Currently every branch of the body assigns a value to this value but one might
worry about future edits.